paasta_tools/kubernetes/application/tools.py

```diff
@@ -31,45 +34,57 @@ def is_valid_application(deployment: V1Deployment):
     return is_valid
 
 
-def list_namespaced_deployments(
-    kube_client: KubeClient, namespace: str, **kwargs
-) -> Sequence[DeploymentWrapper]:
-    return [
-        DeploymentWrapper(deployment)
-        for deployment in kube_client.deployments.list_namespaced_deployment(
-            namespace, **kwargs
-        ).items
-        if is_valid_application(deployment)
-    ]
+def list_paasta_managed_deployments(
+    kube_client: KubeClient, **kwargs
+) -> Dict[Tuple[str, str], List[Application]]:
+    deployments: Dict[Tuple[str, str], List[Application]] = {}
+    for deployment in kube_client.deployments.list_deployment_for_all_namespaces(
+        label_selector="paasta.yelp.com/managed", **kwargs
```

Comment:
for consistency, it might be good to use the paasta_prefixed function to generate the label selector so that we always create labels the same way everywhere